tasks/build_tags.py

```diff
@@ -29,75 +32,200 @@
+])
+
+# IOT_AGENT_TAGS lists the tags needed when building the IoT agent
+IOT_AGENT_TAGS = set([
+    "systemd",
+    "zlib",
+])
+
+# PROCESS_AGENT_TAGS lists the tags necessary to build the process-agent
+PROCESS_AGENT_TAGS = AGENT_TAGS.union(set([
+    "clusterchecks",
     "fargateprocess",
     "orchestrator",
-]
+]))
+
+# SECURITY_AGENT_TAGS lists the tags necessary to build the security agent
+SECURITY_AGENT_TAGS = AGENT_TAGS.union(set([
+    "clusterchecks",
+]))
+
+# PROCESS_AGENT_TAGS lists the tags necessary to build system-probe
+SYSTEM_PROBE_TAGS = AGENT_TAGS.union(set([
+    "clusterchecks",
+    "linux_bpf",
+]))
+
+# TRACE_AGENT_TAGS lists the tags that have to be added when the trace-agent
+TRACE_AGENT_TAGS = set([
+    "docker",
+    "kubeapiserver",
+    "kubelet",
+    "netcgo",
+    "secrets",
+])
+
+# TEST_TAGS lists the tags that have to be added to run tests
+TEST_TAGS = AGENT_TAGS.union(set([
+    "clusterchecks",
+]))
+
+### Tag exclusion lists
 
-LINUX_ONLY_TAGS = [
+# List of tags to always remove when not building on Linux
+LINUX_ONLY_TAGS = set([
     "containerd",
     "cri",
     "netcgo",
     "systemd",
-]
-WINDOWS_32BIT_EXCLUDE_TAGS = [
-    "orchestrator",
+])
+
+# List of tags to always remove when building on Windows
+WINDOWS_EXCLUDE_TAGS = set([
+    "linux_bpf"
+])
+
+# List of tags to always remove when building on Windows 32-bits
+WINDOWS_32BIT_EXCLUDE_TAGS = set([
     "docker",
     "kubeapiserver",
     "kubelet",
-]
-
+    "orchestrator",
+])
 
-def get_default_build_tags(iot=False, process=False, arch="x64", android=False):
+def get_default_build_tags(build="agent", arch="x64"):
     """
-    Build the default list of tags based on the current platform.
+    Build the default list of tags based on the build type and current platform.
 
     The container integrations are currently only supported on Linux, disabling on
     the Windows and Darwin builds.
     """
-    include = ["all"]
-    if iot:
+    # Build setups
+    if build == "agent":
+        include = AGENT_TAGS
+    elif build == "android":
+        include = ANDROID_TAGS
+    elif build == "cluster-agent":
+        include = CLUSTER_AGENT_TAGS
+    elif build == "cluster-agent-cloudfoundry":
+        include = CLUSTER_AGENT_CLOUDFOUNDRY_TAGS
+    elif build == "dogstatsd":
+        include = DOGSTATSD_TAGS
+    elif build == "iot":
         include = IOT_AGENT_TAGS
+    elif build == "process-agent":
+        include = PROCESS_AGENT_TAGS
+    elif build == "security-agent":
+        include = SECURITY_AGENT_TAGS
+    elif build == "system-probe":
+        include = SYSTEM_PROBE_TAGS
+    elif build == "trace-agent":
+        include = TRACE_AGENT_TAGS
+    # Test setups
+    elif build == "test":
+        include = TEST_TAGS
+    elif build == "test-with-process-tags":
+        include = TEST_TAGS.union(PROCESS_AGENT_TAGS)
+    else:
+        print("Warning: unrecognized build type, no build tags included.")
+        include = set()
```

Comment:
nit: maybe a bit more pythonic
nit2: define it on top level
```suggestion
    build_tags = {
        "agent": AGENT_TAGS,
        "android": ANDROID_TAGS,
        # TODO
        "test-with-process-tags": TEST_TAGS.union(PROCESS_AGENT_TAGS),
    }
    include = build_tags.get(build)
    if include is None:
        print("Warning: unrecognized build type, no build tags included.")
        include = set()
```